### PR TITLE
Bold rails upgrade experiment

### PR DESCRIPTION
**Pull request by marktriggs**
:boom:

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -54,4 +55,7 @@ addons:
     - $( ls /var/tmp/*.html | tr "\n" ":" )
     - $( ls /var/log/mysql/* | tr "\n" ":" )
     - $( ls  /var/tmp/aspace-integration.log | tr "\n" ":" )
-sudo: false
+sudo: true
```
Comment (marktriggs) [inline .travis.yml]: Add comment: Set sudo to true so Travis runs us in a VM, not a container

File: backend/Gemfile [hunk 2]
```diff
@@ -45,5 +45,5 @@ gem "zip-zip", "0.3"
 
 
 # Allow plugins to provide their own Gemfiles too.
-require 'asutils'
-ASUtils.load_plugin_gems(self)
+# require 'asutils'
```
Comment (marktriggs) [inline backend/Gemfile]: Need to uncomment this

File: common/asutils.rb [hunk 3]
```diff
@@ -79,11 +79,16 @@ def self.to_json(obj, opts = {})
 
 
   def self.find_base_directory(root = nil)
-    [java.lang.System.get_property("ASPACE_LAUNCHER_BASE"),
+    # JRuby 9K seems to be adding this strange suffix...
+    this_dir = __dir__.gsub(/uri:classloader:\z/, '')
```
Comment (marktriggs) [inline common/asutils.rb]: Check this and write a better comment

File: common/config/config-distribution.rb [hunk 4]
```diff
@@ -64,19 +64,23 @@ def self.get_preferred_config_path
...
-      File.join(File.dirname(__FILE__), "config.rb")
+      File.join(get_devserver_base, "config", "config.rb")
     else
       File.join(Dir.home, ".aspace_config.rb")
     end
 
   end
 
+  def self.get_devserver_base
+    File.dirname(__dir__.gsub(/uri:classloader:.*/, ''))
```
Comment (marktriggs) [inline common/config/config-distribution.rb]: Check this too

File: frontend/Gemfile [hunk 5]
```diff
@@ -1,54 +1,53 @@
...
 
 gem "rubyzip", "1.0.0"
 gem "zip-zip", "0.3"
 
-gem "nokogiri", '~> 1.6.1'
+gem "nokogiri", "1.7.0.1"
+
 
-require 'asutils'
+# require 'asutils'
```
Comment (marktriggs) [inline frontend/Gemfile]: turn this back on

File: frontend/app/helpers/aspace_form_helper.rb [hunk 6]
```diff
@@ -280,10 +280,10 @@ def select(name, options, opts = {})
         opts[:class] = "form-control"
       end
 
+      # obj is [0, entry] instead of entry
```
Comment (marktriggs) [inline frontend/app/helpers/aspace_form_helper.rb]: remove this

File: frontend/config/environments/production.rb [hunk 7]
```diff
@@ -52,27 +66,11 @@
...
-                                file =~ /jquery.tablesorter\/.*/ or 
-                                file =~ /bootstrap-select\/.*/
-                              }]
-
   # Disable delivery errors, bad email addresses will be ignored
   # config.action_mailer.raise_delivery_errors = false
 
   # Enable threaded mode
-  config.threadsafe!
+  # config.threadsafe!
```
Comment (marktriggs) [inline frontend/config/environments/production.rb]: get rid of commented out stuff

File: frontend/config/initializers/disable_xml_params.rb [hunk 8]
```diff
@@ -1 +1 @@
-ActionDispatch::ParamsParser::DEFAULT_PARSERS.delete(Mime::XML)
+# THINKME - old security fix? #ActionDispatch::ParamsParser::DEFAULT_PARSERS.delete(Mime::XML)
```
Comment (marktriggs) [inline frontend/config/initializers/disable_xml_params.rb]: This can go

File: frontend/config/initializers/set_json_backend.rb [hunk 9]
```diff
@@ -1,2 +1,2 @@
 # CVE-2013-0333
-ActiveSupport::JSON.backend = "json_gem"
+# THINKME - obsolete? # ActiveSupport::JSON.backend = "json_gem"
```
Comment (marktriggs) [inline frontend/config/initializers/set_json_backend.rb]: Also gone...

File: frontend/config/initializers/tagged_logging.rb [hunk 10]
```diff
@@ -1,13 +1,14 @@
...
-    def backlog
-      if @logger.respond_to?(:backlog_and_flush)
-        @logger.backlog_and_flush
-      end
-    end
-
-
-  end
-end
+## FIXME: eh?
```
Comment (marktriggs) [inline frontend/config/initializers/tagged_logging.rb]: Nuke it

**Comment by payten** [comment]
🌋 🔥 🐩

**Pull request merged**